core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -1,46 +1,136 @@
 package nl.nn.adapterframework.doc.model;
 
+import java.io.IOException;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.StringTokenizer;
 import java.util.Map.Entry;
 import java.util.stream.Collectors;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.springframework.core.annotation.AnnotationUtils;
+import org.xml.sax.SAXException;
 
 import lombok.Getter;
 import lombok.Setter;
+import nl.nn.adapterframework.configuration.digester.DigesterRule;
+import nl.nn.adapterframework.configuration.digester.DigesterRulesHandler;
+import nl.nn.adapterframework.core.Resource;
 import nl.nn.adapterframework.doc.IbisDoc;
 import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.doc.Utils;
 import nl.nn.adapterframework.util.LogUtil;
+import nl.nn.adapterframework.util.XmlUtils;
 
 public class FrankDocModel {
-	private static final String JAVA_STRING = "java.lang.String";
-
 	private static Logger log = LogUtil.getLogger(FrankDocModel.class);
 
+	private @Getter Map<String, ConfigChildSetterDescriptor> configChildDescriptors;
 	private @Getter List<FrankDocGroup> groups;
 	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+	private @Getter Map<String, ElementType> allTypes = new HashMap<>();
 
 	public FrankDocModel() {
+		configChildDescriptors = new HashMap<>();
 		groups = new ArrayList<>();
 	}
 
```

Comment:
Het lijkt me goed als je ook nog een  method maakt zoals 'populateModel()', die het model vult in de volgorde waarin dat hoort. Als ik het goed heb is dat:
1) configChildDescriptors bepalen uit digesterRules
2) elementTypes creeren uit configChildDescriptors
3) SpringBeans ophalen voor alle elementTypes
4) FrankElements en Attributes creeren uit SpringBeans

Reply:
Ik snap het na ons telefoongesprek van daarnet. Als ik een FrankElement maak voor bijvoorbeeld `<Module>`, dan ontstaan alle relevante types vanzelf vanwege hierarchie van config children. Dat kan ik inderdaad nu al toevoegen.

Reply:
Gedaan. Ik heb de code voor het parsen van @IbisDocRef en @IbisDoc voor attributes wel lelijker moeten maken. Soms wijzen @IbisDocRef annotaties naar methods die er niet zijn. Daarvoor heb ik warnings toegevoegd. Ik stel voor dat we voor een volgende pull request naar alle warnings gaan kijken om te zien waar annotaties missen.